includes/specials/SpecialUserrights.php

```diff
@@ -392,7 +427,7 @@ function switchForm() {
 	 * @return Array:  Tuple of addable, then removable groups
 	 */
 	protected function splitGroups( $groups ) {
-		list( $addable, $removable, $addself, $removeself ) = array_values( $this->changeableGroups() );
+		list( $addable, $removable, $addself, $removeself ) = array_values( $this->getUser()->changeableGroups() );
```

Comment:
The whole code in changeableGroups in this class was a repetition of logic inside of the user class (which I moved to user permissions service), so I deleted the local copy and replaced it with a call to the central definition.